src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/ComponentAccessibilityCodeActionProvider.cs

```diff
@@ -0,0 +1,219 @@
+            FilePathNormalizer filePathNormalizer)
+        {
+            _tagHelperFactsService = tagHelperFactsService ?? throw new ArgumentNullException(nameof(tagHelperFactsService));
+            _filePathNormalizer = filePathNormalizer ?? throw new ArgumentNullException(nameof(filePathNormalizer));
+        }
+
+        override public Task<CommandOrCodeActionContainer> ProvideAsync(RazorCodeActionContext context, CancellationToken cancellationToken)
+        {
+            var commandOrCodeActions = new List<CommandOrCodeAction>();
+
+            // Locate cursor
+            var change = new SourceChange(context.Location.AbsoluteIndex, length: 0, newText: string.Empty);
+            var node = context.CodeDocument.GetSyntaxTree().Root.LocateOwner(change);
+            if (node is null)
+            {
+                return Task.FromResult<CommandOrCodeActionContainer>(null);
+            }
+
+            // Find start tag
+            var startTag = (MarkupStartTagSyntax)node.Ancestors().FirstOrDefault(n => n is MarkupStartTagSyntax);
+            if (startTag == null)
+            {
+                return Task.FromResult<CommandOrCodeActionContainer>(null);
+            }
+
+            // Ignore if has dots, as we only handle short tags
+            if (startTag.Name.Content.Contains("."))
+            {
+                return Task.FromResult<CommandOrCodeActionContainer>(null);
+            }
+
+            if (IsTagUnknown(startTag, context))
+            {
+                AddCreateComponentFromTag(context, startTag, commandOrCodeActions);
+                AddComponentAccessFromTag(context, startTag, commandOrCodeActions);
+            }
+
+            return Task.FromResult(new CommandOrCodeActionContainer(commandOrCodeActions));
+        }
+
+        private void AddCreateComponentFromTag(RazorCodeActionContext context, MarkupStartTagSyntax startTag, List<CommandOrCodeAction> container)
+        {
+            var path = _filePathNormalizer.Normalize(context.Request.TextDocument.Uri.GetAbsoluteOrUNCPath());
+            var newComponentPath = Path.Combine(Path.GetDirectoryName(path), $"{startTag.Name.Content}.razor");
+            if (File.Exists(newComponentPath))
+            {
+                return;
+            }
+
+            var actionParams = new CreateComponentCodeActionParams
+            {
+                Uri = context.Request.TextDocument.Uri,
+                Path = newComponentPath,
+            };
+            var data = JObject.FromObject(actionParams);
+
+            var resolutionParams = new RazorCodeActionResolutionParams
+            {
+                Action = LanguageServerConstants.CodeActions.CreateComponentFromTag,
+                Data = data,
+            };
+            var serializedParams = JToken.FromObject(resolutionParams);
+            var arguments = new JArray(serializedParams);
+
+            container.Add(new CommandOrCodeAction(new Command
```

Comment:
Since the other two are template strings I'm gonna leave the title, but good call on the command name; there's already a constant for it in the LanguageServerConstants